askama_shared/src/lib.rs

```diff
@@ -19,6 +19,47 @@ pub use generator::generate;
 mod generator;
 mod parser;
 
+use std::fmt::{self, Display, Formatter};
+
+pub enum MarkupDisplay<'a, T> where T: 'a + Display {
+    Safe(&'a T),
+    Unsafe(&'a T),
+}
+
+impl<'a, T> MarkupDisplay<'a, T> where T: 'a + Display {
+    pub fn mark_safe(&mut self) {
+        *self = match *self {
+            MarkupDisplay::Unsafe(t) => MarkupDisplay::Safe(t),
+            _ => { return; },
+        }
+    }
+}
+
+impl<'a, T> From<&'a T> for MarkupDisplay<'a, T> where T: 'a + Display {
+    fn from(t: &'a T) -> MarkupDisplay<'a, T> {
+        MarkupDisplay::Unsafe(t)
+    }
+}
+
+impl<'a, T> From<usize> for MarkupDisplay<'a, T> where T: 'a + Display {
+    fn from(t: usize) -> MarkupDisplay<'a, T> {
+        MarkupDisplay::Unsafe(t)
```

Comment:
should this be `Safe` instead?

Reply:
It should, but this won't work anyway -- because we don't have specialization on stable.